pkg/ipmi/oem.go

```diff
@@ -0,0 +1,302 @@
+// Copyright 2017-2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package ipmi
+
+import (
+	"bytes"
+	"encoding/binary"
+	"strconv"
+	"strings"
+	"unsafe"
+
+	"github.com/u-root/u-root/pkg/smbios"
+)
+
+const (
+	_IPMI_FB_OEM_NET_FUNCTION = 0x36
+
+	//OEM Commands
+	_FB_OEM_SET_PROC_INFO = 0x10
+	_FB_OEM_SET_DIMM_INFO = 0x12
+)
+
+type header struct {
+	ManufacturerID    [3]uint8
+	Index             uint8
+	ParameterSelector uint8
+}
+
+type oemData interface {
+	info()
+}
+
+type oemProcProductName struct {
+	header
+	ProductName [48]byte
+}
+
+func (ProcProductName *oemProcProductName) info() {}
+
+type oemProcBasicInfo struct {
+	header
+	CoreNumber            uint8
+	ThreadNumberLSB       uint8
+	ThreadNumberMSB       uint8
+	ProcessorFrequencyLSB uint8
+	ProcessorFrequencyMSB uint8
+	Revision1             uint8
+	Revision2             uint8
+}
+
+func (ProcBasicInfo *oemProcBasicInfo) info() {}
+
+type oemDimmLocation struct {
+	header
+	DIMMPresent   uint8
+	NodeNumber    uint8
+	ChannelNumber uint8
+	DIMMNumber    uint8
+}
+
+func (DimmDataParam1 *oemDimmLocation) info() {}
+
```

Comment:
How about doing something similar to `Event` in https://github.com/u-root/u-root/blob/master/pkg/ipmi/ipmi.go#L136

Example:

```
type Dimm struct {
	ManufacturerID    [3]uint8
	Index             uint8
	CoreNumber            uint8
	ThreadNumberLSB       uint8
	ThreadNumberMSB       uint8
	ProcessorFrequencyLSB uint8
	ProcessorFrequencyMSB uint8
	Revision1             uint8
	Revision2             uint8
}

func (d *Dimm) marshall(param int) ([]byte) {

}
```

So have all the dimm info in a struct and a method to return the bytes depending on the paramenter number.

In that way you will avoid:
* Having an unused info() method
* Setting several times the boardManufacturerID, index on each of the structs

Let me know if it makes sense.

Reply:
Done.